### ISSUE TITLE
reward ai for choosing an effective move

### ISSUE THREAD

**Issue by bulldog98** [PWhiddy/PokemonRedExperiments#58]
I think a good way to improve the ai would be to add an reward for using an effective move, so it hopefully learn to use the types of moves to it's advantage.

**Comment by PWhiddy** [comment]
Yeah! I wonder how that could be detected. This is one of the main resources I used when trying to extract what's happening in the game:
https://datacrystal.romhacking.net/wiki/Pok%C3%A9mon_Red/Blue:RAM_map
I wonder if any of that info could be used to detect that. One could also try testing the pixels of the screen itself but that might be a bit expensive to do every step of the game.

**Comment by dukemagus** [comment]
For that to happen smoothly, it would have to be hardcoded and with a trigger (probably the "it's super effective" multiplying any score gain from inside battles), but that ensures that the program will never deduce this via emergent behavior (it wouldn't need to anymore)

**Comment by MorkyBork** [comment]
> For that to happen smoothly, it would have to be hardcoded and with a trigger (probably the "it's super effective" multiplying any score gain from inside battles), but that ensures that the program will never deduce this via emergent behavior (it wouldn't need to anymore)

This sadly won't work correctly in gen 1 as the text is glitched due to a bug in the messaging display. https://bulbapedia.bulbagarden.net/wiki/Dual-type_damage_misinformation

**Comment by astelmach01** [comment]
Instead of trying to see if it's effective, could possibly take into account how much damage our pokemon did with a move. Also incorporate opponent's level (or stats), since you would always one shot a pokemon that's many levels lower than you
